src/db/deletion/plan.ts

```diff
@@ -0,0 +1,145 @@
+    where: { planId },
+  });
+  if (planProjects.length) {
+    throw new UnprocessableEntityError(
+      `Plan with ID ${planId} has projects, cannot delete`
+    );
+  }
+
+  // Abort if plan or its plan/governing entities are linked to any flow
+  const planEntities = await database.planEntity.find({
+    where: { planId },
+  });
+  const governingEntities = await database.governingEntity.find({
+    where: { planId },
+  });
+  const flows = await database.flowObject.find({
+    where: {
+      [database.Cond.OR]: [
+        {
+          objectType: 'plan',
+          objectID: planId,
+        },
+        {
+          objectType: 'governingEntity',
+          objectID: {
+            [database.Op.IN]: governingEntities.map((ge) => ge.id),
+          },
+        },
+        {
+          objectType: 'planEntity',
+          objectID: {
+            [database.Op.IN]: planEntities.map((pe) => pe.id),
+          },
+        },
+      ],
+    },
+  });
+  if (flows.length) {
+    throw new UnprocessableEntityError(
+      `Plan with ID ${planId} or its entities are attached to flow(s), cannot delete`
+    );
+  }
+};
+
+export const deletePlanById = async (
+  database: Database,
+  planToDelete: PlanId,
+  authGrantRevoker: ParticipantId,
+  trx: Knex.Transaction<any, any>
+) => {
+  const plan = await database.plan.get(planToDelete);
+  if (!plan) {
+    throw new NotFoundError(`Plan with ID ${planToDelete} does not exist`);
+  }
+
+  // Unlink plan categories
+  await database.categoryRef.destroy({
+    where: {
+      objectType: 'plan',
+      objectID: planToDelete,
+    },
+    trx,
+  });
+
+  // Clean up all references in auth tables
+  await deleteAuthTarget(
+    database,
+    { type: 'plan', targetId: planToDelete },
+    authGrantRevoker,
+    trx
+  );
+
+  /**
+   * Related tables are purged through `ON DELETE CASCADE` foreign key constraints:
+   *
+   * `conditionField`
+   *   - `conditionFieldReliesOn`
+   *   - `procedureSectionField`
+   * `procedureSection`
+   *   - `procedureSectionField`
+   * `planLocation`
+   * `attachmentPrototype`
+   * `attachment`
+   *   - `attachmentVersion`
+   *     - `projectVersionAttachment`
+   *   - `measurement`
+   *   - `projectVersionAttachment`
+   * `planYear`
+   * `planEmergency`
+   * `planTag`
+   * `disaggregationCategoryGroup`
+   *   - `disaggregationCategory`
+   * `disaggregationModel`
+   * `entityPrototype`
+   *   - `procedureEntityPrototype`
+   *   - `planEntity`
+   *     - `planEntityVersion`
+   *   - `governingEntity`
+   *     - `governingEntityVersion`
+   *     - `globalClusterAssociation`
+   * `planEntity`
+   *   - `planEntityVersion`
+   * `governingEntity`
+   *   - `governingEntityVersion`
+   *   - `globalClusterAssociation`
+   * `procedureEntityPrototype`
+   * `workflowStatusOption`
+   *   - `workflowStatusOptionStep`
+   * `planReportingPeriod`
+   *   - `measurement`
+   *     - `measurementVersion`
+   * `planVersion`
```

Comment:
Recently added **planVersionOrganization** table that you might want to include in this doc.

```suggestion
   * `planVersion`
   *   - `planVersionOrganization`
```

Reply:
You are right and I have realized how terrible this is from maintenance standpoint. It will likely become outdated with any change.